### ISSUE TITLE
File saves with incorrect style

### ISSUE THREAD

**Issue by MasterSlynx** [ClosedXML/ClosedXML#1167]
**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**

e.g. 0.94.2

**What is the current behavior?**

One of the cells is saved with incorrect borders style

**What is the expected behavior or new feature?**

Style should be same as in original opened file.

**Did this work in previous versions of our tool?  Which versions?**

Yes, v0.93.1

## Reproducibility

**Code to reproduce problem:**
```c#
public void Main()
{
            var wb = new XLWorkbook(@"C:\Test\2\XLTest.xlsx");
            wb.SaveAs(@"C:\Test\2\XLTest4.xlsx");
}
```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[XLTest.xlsx](https://github.com/ClosedXML/ClosedXML/files/3024444/XLTest.xlsx)

**Comment by MasterSlynx** [comment]
I think problem may be related to "TEST" spreadsheet that gets removed before saving. There was a reference in of the the cells (C7) for data validation. Looks like old version saves it regardles of that but version 0.94.2 throws exception before saving. That's just an assumption.
However, the data validation (drop down list) in some of the cells is still missing and the "MONTHLY TOTALS" cell contains some weird black borders that wasn't there before, both in template and generated file in previous version.
It's a bit late (01:40AM), so maybe I'm missing something but I will be checking everything again tomorrow, maybe it's a bug in my code.

**Comment by MasterSlynx** [comment]
Right.... data validation seems to be working fine in 0.94.2. Looks like cells range was wrong - at least in 0.94.2 😅 
Last weird thing is the black borders in one of the cells.... needs to check it carefully too....

**Comment by igitur** [comment]
Please reduce your code to the smallest, simplest version that reproduces the problem.

**Comment by MasterSlynx** [comment]
Well, now when saving and data validation is not a problem, code can be a little simplier... 🙄

**Comment by MasterSlynx** [comment]
Ok, I've actualy cloned the current version to check if the error is still there and it's not working.
I've spent few minutes and looked closely what causes the issue. 
The code for merging cells doesn't work properly. The problematic lines are 393 and 394 in XLRangeBase.Merge(Boolean) method.
```c#
if (cellsUsed.Any(c => (c.Style as XLStyle).Key.Border != defaultStyle.Border))
    asRange.Style.Border.SetInsideBorder(XLBorderStyleValues.None);
```
This method creates 5 additonal (in my specific example) styles in result file (styles.xml). Fully black, just with different border styles (thin,double).
Removing these lines helps but breaks the MergedRangesTests.UnmergedCellsPreserveStyle test, so I will leave it to someone who have more in-depth knowledge on how it should work.

**Comment by Pankraty** [comment]
This was fixed by #1137.
